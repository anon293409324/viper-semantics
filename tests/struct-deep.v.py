s: {f01a:num,
    f01b:{f02a:num,
    f02b:{f03a:num,
    f03b:{f04a:num,
    f04b:{f05a:num,
    f05b:{f06a:num,
    f06b:{f07a:num,
    f07b:{f08a:num,
    f08b:{f09a:num,
    f09b:{f10a:num,
    f10b:{f11a:num,
    f11b:{f12a:num,
    f12b:{f13a:num,
    f13b:{f14a:num,
    f14b:{f15a:num,
    f15b:{f16a:num,
    f16b:{f17a:num,
    f17b:{f18a:num,
    f18b:{f19a:num,
    f19b:num
   }}}}}}}}}}}}}}}}}}}
def foo(x:num):
    self.s = {f01a:x,
              f01b:{f02a:x,
              f02b:{f03a:x,
              f03b:{f04a:x,
              f04b:{f05a:x,
              f05b:{f06a:x,
              f06b:{f07a:x,
              f07b:{f08a:x,
              f08b:{f09a:x,
              f09b:{f10a:x,
              f10b:{f11a:x,
              f11b:{f12a:x,
              f12b:{f13a:x,
              f13b:{f14a:x,
              f14b:{f15a:x,
              f15b:{f16a:x,
              f16b:{f17a:x,
              f17b:{f18a:x,
              f18b:{f19a:x,
              f19b:x
             }}}}}}}}}}}}}}}}}}}
