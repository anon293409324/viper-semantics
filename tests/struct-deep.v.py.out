[seq,
  [return,
    0,
    [lll,
      [seq,
        [seq,
          [mstore, 28, [calldataload, 0]],
          [mstore, 32, 1461501637330902918203684832716283019655932542976],
          [mstore, 64, 170141183460469231731687303715884105727],
          [mstore, 96, -170141183460469231731687303715884105728],
          [mstore, 128, 1701411834604692317316873037158841057270000000000],
          [mstore, 160, -1701411834604692317316873037158841057280000000000]],
        # Line 22
        [if,
          [eq, [mload, 0], 3650092561 <foo>],
          [seq,
            [calldatacopy, 320, 4, 32],
            [assert, [iszero, callvalue]],
            /* checking num input */ 
            [clamp,
              [mload, 96],
              [calldataload, [add, 4, 0]],
              [mload, 64]],
            # Line 23
            [with,
              _L,
              [sha3_32, 0 <self.s>],
              [seq,
                [sstore, [add, _L, 0 <f01a>], [mload, 320 <x>]],
                [with,
                  _L,
                  [sha3_32, [add, _L, 1 <f01b>]],
                  [seq,
                    [sstore, [add, _L, 0 <f02a>], [mload, 320 <x>]],
                    [with,
                      _L,
                      [sha3_32, [add, _L, 1 <f02b>]],
                      [seq,
                        [sstore, [add, _L, 0 <f03a>], [mload, 320 <x>]],
                        [with,
                          _L,
                          [sha3_32, [add, _L, 1 <f03b>]],
                          [seq,
                            [sstore, [add, _L, 0 <f04a>], [mload, 320 <x>]],
                            [with,
                              _L,
                              [sha3_32, [add, _L, 1 <f04b>]],
                              [seq,
                                [sstore, [add, _L, 0 <f05a>], [mload, 320 <x>]],
                                [with,
                                  _L,
                                  [sha3_32, [add, _L, 1 <f05b>]],
                                  [seq,
                                    [sstore, [add, _L, 0 <f06a>], [mload, 320 <x>]],
                                    [with,
                                      _L,
                                      [sha3_32, [add, _L, 1 <f06b>]],
                                      [seq,
                                        [sstore, [add, _L, 0 <f07a>], [mload, 320 <x>]],
                                        [with,
                                          _L,
                                          [sha3_32, [add, _L, 1 <f07b>]],
                                          [seq,
                                            [sstore, [add, _L, 0 <f08a>], [mload, 320 <x>]],
                                            [with,
                                              _L,
                                              [sha3_32, [add, _L, 1 <f08b>]],
                                              [seq,
                                                [sstore, [add, _L, 0 <f09a>], [mload, 320 <x>]],
                                                [with,
                                                  _L,
                                                  [sha3_32, [add, _L, 1 <f09b>]],
                                                  [seq,
                                                    [sstore, [add, _L, 0 <f10a>], [mload, 320 <x>]],
                                                    [with,
                                                      _L,
                                                      [sha3_32, [add, _L, 1 <f10b>]],
                                                      [seq,
                                                        [sstore, [add, _L, 0 <f11a>], [mload, 320 <x>]],
                                                        [with,
                                                          _L,
                                                          [sha3_32, [add, _L, 1 <f11b>]],
                                                          [seq,
                                                            [sstore, [add, _L, 0 <f12a>], [mload, 320 <x>]],
                                                            [with,
                                                              _L,
                                                              [sha3_32, [add, _L, 1 <f12b>]],
                                                              [seq,
                                                                [sstore, [add, _L, 0 <f13a>], [mload, 320 <x>]],
                                                                [with,
                                                                  _L,
                                                                  [sha3_32, [add, _L, 1 <f13b>]],
                                                                  [seq,
                                                                    [sstore, [add, _L, 0 <f14a>], [mload, 320 <x>]],
                                                                    [with,
                                                                      _L,
                                                                      [sha3_32, [add, _L, 1 <f14b>]],
                                                                      [seq,
                                                                        [sstore, [add, _L, 0 <f15a>], [mload, 320 <x>]],
                                                                        [with,
                                                                          _L,
                                                                          [sha3_32, [add, _L, 1 <f15b>]],
                                                                          [seq,
                                                                            [sstore, [add, _L, 0 <f16a>], [mload, 320 <x>]],
                                                                            [with,
                                                                              _L,
                                                                              [sha3_32, [add, _L, 1 <f16b>]],
                                                                              [seq,
                                                                                [sstore, [add, _L, 0 <f17a>], [mload, 320 <x>]],
                                                                                [with,
                                                                                  _L,
                                                                                  [sha3_32, [add, _L, 1 <f17b>]],
                                                                                  [seq,
                                                                                    [sstore, [add, _L, 0 <f18a>], [mload, 320 <x>]],
                                                                                    [with,
                                                                                      _L,
                                                                                      [sha3_32, [add, _L, 1 <f18b>]],
                                                                                      [seq,
                                                                                        [sstore, [add, _L, 0 <f19a>], [mload, 320 <x>]],
                                                                                        [sstore, [add, _L, 1 <f19b>], [mload, 320 <x>]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]]],
            # Line 22
            stop]]],
      0]]]
['_sym_2', 'JUMP', '_sym_1', 'BLANK', ['PUSH1', 0, 'CALLDATALOAD', 'PUSH1', 28, 'MSTORE', 'PUSH21', 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'PUSH1', 32, 'MSTORE', 'PUSH16', 127, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 'PUSH1', 64, 'MSTORE', 'PUSH32', 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'PUSH1', 96, 'MSTORE', 'PUSH21', 1, 42, 5, 241, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 253, 171, 244, 28, 0, 'PUSH1', 128, 'MSTORE', 'PUSH32', 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 254, 213, 250, 14, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'PUSH1', 160, 'MSTORE', 'PUSH4', 217, 143, 254, 17, 'PUSH1', 0, 'MLOAD', 'EQ', 'ISZERO', '_sym_3', 'JUMPI', 'PUSH1', 32, 'PUSH1', 4, 'PUSH2', 1, 64, 'CALLDATACOPY', 'CALLVALUE', 'ISZERO', 'ISZERO', 'PC', 'JUMPI', 'PUSH1', 96, 'MLOAD', 'PUSH1', 0, 'PUSH1', 4, 'ADD', 'CALLDATALOAD', 'DUP1', 'PUSH1', 64, 'MLOAD', 'SWAP1', 'SGT', 'PC', 'JUMPI', 'DUP1', 'SWAP2', 'SWAP1', 'SLT', 'PC', 'JUMPI', 'POP', 'PUSH1', 0, 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH1', 1, 'DUP2', 'ADD', 'PUSH1', 192, 'MSTORE', 'PUSH1', 192, 'PUSH1', 32, 'SHA3', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 0, 'DUP3', 'ADD', 'SSTORE', 'PUSH2', 1, 64, 'MLOAD', 'PUSH1', 1, 'DUP3', 'ADD', 'SSTORE', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'POP', 'STOP', '_sym_3', 'JUMPDEST'], '_sym_2', 'JUMPDEST', '_sym_1', '_sym_2', 'SUB', '_sym_1', 'PUSH1', 0, 'CODECOPY', '_sym_1', '_sym_2', 'SUB', 'PUSH1', 0, 'RETURN']
0x61027656600035601c52740100000000000000000000000000000000000000006020526f7fffffffffffffffffffffffffffffff6040527fffffffffffffffffffffffffffffffff8000000000000000000000000000000060605274012a05f1fffffffffffffffffffffffffdabf41c006080527ffffffffffffffffffffffffed5fa0e000000000000000000000000000000000060a05263d98ffe1160005114156102715760206004610140373415155857606051600060040135806040519013585780919012585750600060c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c06020206101405160008201556001810160c05260c060202061014051600082015561014051600182015550505050505050505050505050505050505050005b5b61000461027603610004600039610004610276036000f3
